python/lsst/ts/tunablelaser/csc.py

```diff
@@ -131,14 +151,36 @@ def assert_substate(self, substates, action):
 
     async def handle_summary_state(self):
         if self.disabled_or_enabled:
-            if not self.model.connected:
-                self.model.connect()
+            if self.simulator is None and self.simulation_mode:
+                self.simulator = MockServer()
+                await self.simulator.start_task
+                self.log.debug(f"port={self.simulator.port}")
+                self.model.commander = TCPIPClient(
+                    self.model.config.host,
+                    self.simulator.port,
+                    self.model.config.timeout,
+                )
+                self.model._update_commander()
+                self.log.debug(f"port={self.model.cpu8000.commander.port}")
+            elif not self.simulation_mode:
+                self.model.commander = TCPIPClient(
+                    self.model.config.host,
+                    self.model.config.port,
+                    self.model.config.timeout,
+                )
+                self.model._update_commander()
+                self.log.debug(f"port={self.model.commander.port}")
```

Comment:
I think you have a bug here. It looks to me like you will create a new commander whenever you transition to DISABLED or ENABLED state. Thus in the normal sequence of state transitions you will create 3 commanders (one for STANDBY->DISABLED, one for DISABLED->ENABLED and a final one for ENABLED->DISABLED) when you only need one.

Also I am pretty sure you need host = localhost when connecting to the simulator.

Here is one possibility (untested). I separated the creation of the mock server
from the connection, in an attempt to simplify the logic and reduce repetition.
```
async def handle_summary_state(self):
    if self.disabled_or_enabled:
        if self.simulation_mode:
            if self.simulator is None:
                self.simulator = MockServer()
                await self.simulator.start_task
                # Make sure we aren't connected to anything (paranoia)
                await self.model.disconnect()
            host = tcpip.LOCAL_HOST
            port = self.simulator.port
        else:
            host = self.model.config.host
            port = self.model.config.port
        if not self.connected:
            self.model.commander = TCPIPClient(
                host=host,
                port=port,
                timeout=self.model.config.timeout,
            )
            self.model._update_commander()
            self.log.debug(f"Connecting to port={self.model.commander.port}")
            await self.model.connect()
        if self.telemetry_task.done():
            self.telemetry_task = asyncio.create_task(self.telemetry())
    else:
        if self.model.is_propgating:
            await self.model.stop_propagating()
        if self.simulator is not None:
            await self.simulator.close()
            self.simulator = None
        await self.model.disconnect()
        self.telemetry_task.cancel()
```

Reply:
Also please consider making `connect` a method on your model: a method that takes host and port as arguments and:
* Raises if already connected.
* Creates a new commander (TCP/IP client).
* Calls self._update_commander.

That way it is impossible to create a new commander while forgetting to call model._update_commander`. 
Also it avoids the need for the CSC to call the private method _update_commander method of your model class.